src/main/java/hudson/plugins/throttleconcurrents/ThrottleJobProperty.java

```diff
@@ -218,20 +218,54 @@ public boolean isThrottleMatrixConfigurations() {
                 : ThrottleMatrixProjectOptions.DEFAULT.isThrottleMatrixConfigurations();
     }
 
-    public List<String> getParamsToCompare() {
-        if (paramsToCompare == null) {
-            if ((paramsToUseForLimit != null)) {
-                if ((paramsToUseForLimit.length() > 0)) {
-                    paramsToCompare = Arrays.asList(paramsToUseForLimit.split(","));
-                }
-                else {
-                    paramsToCompare = new ArrayList<String>();
-                }
+    /**
+     * @param paramsToUseForLimit   Set the paramsToCompare list
+     * of Strings from the user-input token-separated String.
+     *
+     * There is no return, but the this.paramsToCompare will be a
+     * valid non-null populated or empty array after this call.
+     */
+    public void setParamsToCompare(String paramsToUseForLimit) {
+        // If there is any contents, tell GC that it is reapable
+        this.paramsToCompare = null;
+
+        if ((paramsToUseForLimit != null)) {
+            if ((paramsToUseForLimit.length() > 0)) {
+                List<String> list = new ArrayList<String>(Arrays.asList(ArrayUtils.nullToEmpty(paramsToUseForLimit.split(PARAMS_LIMIT_SEPARATOR))));
+                // Due to type casting, DO NOT MERGE with the line above
+                // in accordance with https://stackoverflow.com/a/5520808/4715872
+                // Logically, we should only remove all blanks "" and the
+                // split() should have taken care of other chars... but
+                // better safe than sorry. Also remove("") did not seem
+                // to cut it for more than one blank entry in the List.
+                list.removeAll(Arrays.asList("", " ", "\n", "\t", ","));
+                this.paramsToCompare = list;
             }
             else {
-                paramsToCompare = new ArrayList<String>();
+                this.paramsToCompare = new ArrayList<String>();
             }
         }
+        else {
+            this.paramsToCompare = new ArrayList<String>();
```

Comment:
Might be worth using `Collections.emptyList()` while at it.

Reply:
I thought of it while googling for modern patterns, but saw some value in keeping the minimum dependencies needed to build this (adding as few import's as I can).

Would the other solutions be more efficient for example? This is a moderately hot codepath I suppose (executed whenever a job might pass from queued to building).

Reply:
It does not appear to be on a hot code path to me. Though still, `Collections` is part of JDK so the only dependency in play is adding the import line.

Reply:
Oh, ok. I thought it was part of apache tooling.

Reply:
I looked at it and am inclined to leave as is, to keep the code styled the same way and the type-casting clearer in the different conditional cases we have to set up the list contents here.